CMakeLists.txt

```diff
@@ -366,6 +366,15 @@ if (NOT $ENV{S2N_LIBCRYPTO} STREQUAL "awslc")
     target_compile_options(${PROJECT_NAME} PRIVATE -Wcast-qual)
 endif()
 
+if (COVERAGE)
+    # https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html
+    if (NOT ${CMAKE_C_COMPILER_ID} MATCHES Clang)
+        message(FATAL_ERROR "This project requires clang for coverage support".)
+    endif()
```

Comment:
Why? Can you add a comment documenting the reason for this requirement?

Reply:
Source coverage is an LLVM (Clang) specific feature. I'll add a comment along the lines of
```
# Coverage is done using LLVM source based coverage. This is only supported on LLVM compilers.
# GCC would fail with "unrecognized compile options" on -fprofile-instr-generate -fcoverage-mapping
```